app/views/activities/_share_button.html.erb

```diff
@@ -1,4 +1,16 @@
-<%= <div id="fb-root"></div>
+<%# <div id="fb-root"></div>
 <script async defer crossorigin="anonymous" src="https://connect.facebook.net/en_GB/sdk.js#xfbml=1&version=v15.0" nonce="1WMVtyEy"></script>
 <div class="fa fa-share-alt" data-href="https://developers.facebook.com/docs/plugins/" data-layout="button_count" data-size="small"><a target="_blank" href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fdevelopers.facebook.com%2Fdocs%2Fplugins%2F&amp;src=sdkpreparse" class="fb-xfbml-parse-ignore"></a></div> %>
-<span><%= button_to "", "https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fdevelopers.facebook.com%2Fdocs%2Fplugins%2F&amp;src=sdkpreparse", class:" icons border-0 fa fa-share-alt bg-transparent" %></span>
+<%# <span>button_to "", "https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fdevelopers.facebook.com%2Fdocs%2Fplugins%2F&amp;src=sdkpreparse", class:" icons border-0 fa fa-share-alt bg-transparent"</span> %>
+
+<script>
+document.getElementById('shareBtn').onclick = function() {
+  FB.ui({
+    display: 'popup',
+    method: 'share',
+    href: 'https://playsnap.herokuapp.com/activities/<%= @activity %>',
```

Comment:
can try with `http://localhost:3000/` first, and test see what happens

Reply:
any errors on rails server ah?

Reply:
@prawnto are we passing @activity to this share button also?

Reply:
This `<%= render "share_button" %>` is in activity show.rb, never pass the activity value, so cannot this share button file cannot do anything

Reply:
Maybe need to write something like that `<%= render "heart_button", activity: @activity, heart: @heart %>` @prawnto

Reply:
@goh0222 no errors on rails server though as it links to FB. i've amended to <%= render "share_button", activity: @activity %> still same prompt, as FB don't allow testing on localhost

Reply:
href: 'http://localhost:3000/activities/<%= activity.id %>'

Reply:
@prawnto how about this?

Reply:
no change in prompt, didn't work i think

Reply:
ps, try again with this `href: "http://localhost:3000/activities/#{activity.id}"`

Reply:
if cannot, what is the plan? just push to heroku and test isit?

Reply:
still the same .. should i try this? https://samvikshana.weebly.com/blog/how-to-use-facebook-sharing-debugger-and-twitter-card-validator-with-localhost-urls

Reply:
can we try `http://localhost:3000/activities` without id, see if there's something?

Reply:
you can also try out using 'https://developers.facebook.com/docs/' see if there is something

Reply:
still no change

Reply:
i tried this href: "<%= url_for(@activity) %>" according to https://api.rubyonrails.org/classes/ActionView/RoutingUrlFor.html#method-i-url_for

still no. i suspect coz FB only allows testing on live URL

Reply:
this is a live url leh: `https://developers.facebook.com/docs/`

Reply:
![image](https://user-images.githubusercontent.com/100469103/193237891-3484eb80-0ef2-4cc0-a602-5e0994ad7ab3.png)

yea the live url also showed the same prompt.. means the script or button to needs fixing? btw, i tried the localtunnel and got this prompt on the sharer debugger

Reply:
```
<span><%= button_to "",  class:"icons border-0 fa fa-share-alt bg-transparent", id: "shareBtn" %></span>

<script>
document.getElementById('shareBtn').onclick = function() {
  FB.ui({
    display: 'popup',
    method: 'share',
    href: 'https://developers.facebook.com/docs/',
  }, function(response){});
}
</script>
```

Reply:
can try this? @prawnto

Reply:
yaamanni found that initially. need to check where it is from. the block didn't work as button couldn't be displayed

Reply:
need a url in the button_to. i tried to add <%= button_to "", 'https://developers.facebook.com/docs/', class:"icons border-0 fa fa-share-alt bg-transparent", id: "shareBtn" %> and it is now directed to:
![image](https://user-images.githubusercontent.com/100469103/193242263-f5711c13-0335-41a5-aef1-4623016d5e4b.png)

Reply:
``<%= button_to "", 'https://www.facebook.com/dialog/share?app_id=145634995501895&display=popup&href=https%3A%2F%2Fdevelopers.facebook.com%2Fdocs%2F&redirect_uri=https%3A%2F%2Fdevelopers.facebook.com%2Ftools%2Fexplorer', class:"icons border-0 fa fa-share-alt bg-transparent", id: "shareBtn" %>

<script>
document.getElementById('shareBtn').onclick = function() {
  FB.ui({
    display: 'popup',
    method: 'share',
    href: 'https://developers.facebook.com/docs/sharing/reference/share-dialog',
  }, function(response){});
}
</script>``

didn't work too